kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -359,53 +386,148 @@ protected ProcurementCardDefault retrieveProcurementCardDefault(String creditCar
      */
     @Override
     public boolean autoApproveProcurementCardDocuments() {
-        //if auto approve is turned on
-        boolean autoApproveOn = super.parameterService.getParameterValueAsBoolean(
-                ProcurementCardAutoApproveDocumentsStep.class,
-                ProcurementCardDocumentRuleConstants.AUTO_APPROVE_DOCUMENTS_IND);
+        // if auto approve is turned on
+        boolean autoApproveOn = parameterService.getParameterValueAsBoolean(ProcurementCardAutoApproveDocumentsStep.class, ProcurementCardDocumentRuleConstants.AUTO_APPROVE_DOCUMENTS_IND);
 
-        if (!autoApproveOn) { // no auto approve?  then skip out of here...
+        if (!autoApproveOn) { // no auto approve? then skip out of here...
             return true;
         }
 
+        StringBuffer autoApproveErrorStr = new StringBuffer();
+
+        // get Parameter information
+        Timestamp currentDate = dateTimeService.getCurrentTimestamp();
+        int autoApproveNumberDays = Integer.parseInt(parameterService.getParameterValueAsString(ProcurementCardAutoApproveDocumentsStep.class, ProcurementCardDocumentRuleConstants.AUTO_APPROVE_NUMBER_OF_DAYS));
```

Comment:
Misty wasn't available for comment on this. Integer.parseInt will throw an exception if the string passed to it isn't also a number. This is how it works in 3, and there's no default value in the Specs.